iq/iq.go

```diff
@@ -502,17 +520,42 @@ func (i *Server) pollIQServer(statusURL string, finished chan resultError) error
 				Message: "Could not unmarshal response from IQ server",
 			}
 		}
+
+		i.logLady.WithFields(logrus.Fields{
+			"response": response,
+		}).Trace("Nexus IQ polling response")
+
 		statusURLResp = response
 		if response.IsError {
 			finished <- resultError{finished: true, err: nil}
 		}
+
+		i.populateAbsoluteURL()
 		finished <- resultError{finished: true, err: nil}
 	}
 	i.tries++
 	fmt.Print(".")
 	return err
 }
 
+func (i *Server) populateAbsoluteURL() {
+	if !strings.HasPrefix(statusURLResp.ReportHTMLURL, i.Options.Server) {
+		// newer versions of IQ (104+) use relative urls
+
+		iqServerUrl := i.Options.Server
+		// slash madness
+		var pathSlash string
+		if strings.HasPrefix(statusURLResp.ReportHTMLURL, "/") || strings.HasSuffix(iqServerUrl, "/") {
```

Comment:
Don't we protect against the `iqServerUrl` ever having a leading slash?

https://github.com/sonatype-nexus-community/go-sona-types/blob/master/iq/iq.go#L173-L175

I realize you could end up with it if you were constructing stuff yourself (and I applaud for being more cautious than I!), but it makes me wonder if you want to pull the parameter sanitization stuff out of `New` and into another private function or method, so you could call that for utility things like this? SPITBALLIN

Reply:
You might like this:

https://play.golang.org/p/AUeCwD7he2p

May as well just call the trim on both, rather than check etc... (trivial lil thing)

Reply:
Noice! Using the parse method also leaves the door open to more validation (like ResolveReference).
This also exposed a couple more cases I hadn't thunk about.  5ab85ee